pkg/agent/utils_test.go

```diff
@@ -233,4 +233,27 @@ var _ = Describe("Assert ParseCSE", func() {
 		Expect(err).NotTo(BeNil())
 		Expect(err.Error()).To(ContainSubstring("InstanceErrorCode=CSEMessageUnmarshalError"))
 	})
+
+	It("when Windows cse output is correct with exitcode is empty", func() {
+		testMessage := "vmss aks-agentpool-test-vmss instance 0 vmssCSE message : Command execution finished"
+		res, err := ParseCSEMessage(testMessage)
+		Expect(err).To(BeNil())
+		Expect(res.ExitCode).To(Equal("0"))
+		Expect(res.Output).To(Equal("vmss aks-agentpool-test-vmss instance 0 vmssCSE message : Command execution finished"))
+	})
+
+	It("when Windows cse output is correct with exitcode is not empty", func() {
+		testMessage := "vmss aks-agentpool-test-vmss instance 0 vmssCSE message : Command execution finished, but failed because it returned a non-zero exit code of: '5'"
+		res, err := ParseCSEMessage(testMessage)
+		Expect(err).To(BeNil())
+		Expect(res.ExitCode).To(Equal("5"))
+		Expect(res.Output).To(Equal("vmss aks-agentpool-test-vmss instance 0 vmssCSE message : Command execution finished, but failed because it returned a non-zero exit code of: '5'"))
+	})
+
+	It("when Windows cse output is incorrect", func() {
+		testMessage := "vmss aks-agentpool-test-vmss instance 0 vmssCSE message : Command execution succeeded"
+		_, err := ParseCSEMessage(testMessage)
+		Expect(err).NotTo(BeNil())
+		Expect(err.Error()).To(ContainSubstring("InstanceErrorCode=InvalidCSEMessage"))
```

Comment:
It is better to check whether the original message is included in the err.Error().